pkg/forwarder/forwarder_test.go

```diff
@@ -6,278 +6,153 @@
+	assert.Contains(t, transactions[1].Endpoint, "api_key=api-key-2")
+	assert.Contains(t, transactions[2].Endpoint, "api_key=api-key-1")
+	assert.Contains(t, transactions[3].Endpoint, "api_key=api-key-2")
+}
 
-	expectedPayload = []byte("SubmitV1CheckRuns payload")
-	expectedEndpoint = "/api/v1/check_run"
-	expectAPIKeyInQuery = true
-	payloads = []*[]byte{}
-	payloads = append(payloads, &expectedPayload)
-	assert.Nil(t, forwarder.SubmitV1CheckRuns(payloads, map[string]string{"Content-Type": "application/unit-test"}))
-	<-wait
-	<-wait
-	require.Len(t, forwarder.retryQueue, 0)
+func TestSendHTTPTransactions(t *testing.T) {
+	forwarder := NewDefaultForwarder(keysPerDomains)
+	endpoint := "/api/foo"
+	p1 := []byte("A payload")
+	payloads := Payloads{&p1}
+	headers := make(http.Header)
+	tr := forwarder.createHTTPTransactions(endpoint, payloads, false, headers)
+	err := forwarder.sendHTTPTransactions(tr)
 
-	expectedPayload = []byte("SubmitV1Intake payload")
-	expectedEndpoint = "/intake/"
-	expectedHeaders.Set("Content-type", "application/json")
-	expectAPIKeyInQuery = true
-	payloads = []*[]byte{}
-	payloads = append(payloads, &expectedPayload)
-	assert.Nil(t, forwarder.SubmitV1Intake(payloads, map[string]string{"Content-Type": "application/unit-test"}))
-	<-wait
-	<-wait
-	require.Len(t, forwarder.retryQueue, 0)
-}
+	// fw is stopped, we should get an error
+	assert.NotNil(t, err)
 
-func TestSubmitWithProxy(t *testing.T) {
-	targetURL := "http://test:2121"
-	firstKey := "api_key1"
-	wait := make(chan bool)
-
-	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		defer func() { wait <- true }()
-		assert.Equal(t, "POST", r.Method)
-		assert.Equal(t, targetURL, fmt.Sprintf("%s://%s", r.URL.Scheme, r.URL.Host))
-		assert.Equal(t, firstKey, r.Header.Get(apiHTTPHeaderKey))
-
-		w.WriteHeader(http.StatusOK)
-	}))
-	defer ts.Close()
-	config.Datadog.Set("proxy", map[string]interface{}{"http": ts.URL})
-	defer config.Datadog.Set("proxy", nil)
-
-	forwarder := NewDefaultForwarder(map[string][]string{targetURL: []string{firstKey}})
-	// delay next retry cycle so we can inspect retry queue
-	flushInterval = 5 * time.Hour
-	forwarder.NumberOfWorkers = 1
 	forwarder.Start()
-	defer forwarder.Stop()
+	err = forwarder.sendHTTPTransactions(tr)
+	assert.Nil(t, err)
+	forwarder.Stop()
+}
 
-	payload := []byte("SubmitSeries payload")
-	payloads := []*[]byte{}
-	payloads = append(payloads, &payload)
-	assert.Nil(t, forwarder.SubmitSeries(payloads, map[string]string{"Content-Type": "application/unit-test"}))
-	// wait for the queries to complete before changing expected value
-	<-wait
-	require.Len(t, forwarder.retryQueue, 0)
+func TestRequeueTransaction(t *testing.T) {
+	forwarder := NewDefaultForwarder(nil)
+	tr := NewHTTPTransaction()
+	assert.Len(t, forwarder.retryQueue, 0)
+	forwarder.requeueTransaction(tr)
+	assert.Len(t, forwarder.retryQueue, 1)
 }
 
-func TestSubmitWithProxyAndPassword(t *testing.T) {
```

Comment:
I think we could still check the `proxy` settings get correctly applied by asserting:
`forwarder.Worker.Client.Transport.Proxy` without needing to bring up an http server. Not sure if that's necessary since we already have some proxy facility testing in `pkg/util/common_test.go`

Reply:
That's the point, the forwarder relies on the `util` package to setup the http client.
A good test would be ensuring that the above statement is true and doesn't change over time.